From 40f02fef77f9eb2cf2522255ade19fed343f5ff5 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Tue, 26 Jul 2022 12:06:29 +0300
Subject: [PATCH] [SW-97677] gaudi2: enable elbi regardless of rst src

In case of FLR reset and PRSTN reset, elbi access is
disabled in different way though pcie core. But while
enabling ELBI, it can be enabled at both pcie core
and elbi master interface level without checking rst
src.

Change-Id: I75852aab911b70d41fc70e7ed15762e41fd4e409
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 57db88d..bee62c4 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -973,16 +973,11 @@
 #define PCIE_WRAP_OUTBOUND_TIMEOUT_US	100
 void hl_pcie_enable_access(void)
 {
-	unsigned int rst_src = hl_rst_src_get();
-
 	/* Enable the ELBI access */
-	if ((rst_src == HL_FLR_RST) || (rst_src == HL_PRSTN_RST)) {
-		regmap_writel(0, mmPCIE_CORE_REQ_RETRY_EN);
-	} else {
-		regmap_writel(RR_PRIV_RANGE_RESET_VAL,
-				mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0);
-		regmap_writel(0, mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
-	}
+	regmap_writel(RR_PRIV_RANGE_RESET_VAL,
+		      mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0);
+	regmap_writel(0, mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
+	regmap_writel(0, mmPCIE_CORE_REQ_RETRY_EN);
 
 	/* Disable HOST_ACCESS_TERMINATION */
 	regmap_writel(0, mmPCIE_WRAP_HOST_ACCESS_TERMINATION);
